core/core-workflow-steps/src/main/java/org/eclipse/sw360/antenna/workflow/generators/SourceZipWriterImpl.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.workflow.generators;
+
+import org.eclipse.sw360.antenna.api.Attachable;
+import org.eclipse.sw360.antenna.api.IArtifactFilter;
+import org.eclipse.sw360.antenna.api.IAttachable;
+import org.eclipse.sw360.antenna.api.IProcessingReporter;
+import org.eclipse.sw360.antenna.api.exceptions.ExecutionException;
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactFilename;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactSourceFile;
+import org.eclipse.sw360.antenna.model.reporting.MessageType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.zip.*;
+
+public class SourceZipWriterImpl {
+    private static final String IDENTIFIER = "sources-zip";
+    private static final String CLASSIFIER = "antenna-sources-zip";
+    private static final String TYPE = "zip";
+
+    private IArtifactFilter notAllowed;
+    private IArtifactFilter unknownMatchStateFilter;
+    private Path sourceZipPath;
+    private IProcessingReporter reporter;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceZipWriterImpl.class);
+
+    public SourceZipWriterImpl(Path sourceZipPath, IArtifactFilter notAllowed, IArtifactFilter unknownMatchStateFilter, IProcessingReporter reporter) {
+        this.sourceZipPath = sourceZipPath;
+        this.notAllowed = notAllowed;
+        this.unknownMatchStateFilter = unknownMatchStateFilter;
+        this.reporter = reporter;
+    }
+
+    public Map<String, IAttachable> produce(Collection<Artifact> artifacts) {
+        File sourceArchive = createZip(new ArrayList<>(artifacts));
+        return Collections.singletonMap(IDENTIFIER, new Attachable(TYPE, CLASSIFIER, sourceArchive));
```

Comment:
sourceArchive not really needed
```suggestion
        return Collections.singletonMap(IDENTIFIER, new Attachable(TYPE, CLASSIFIER, createZip(new ArrayList<>(artifacts))));
```